### PR TITLE
When converting HEIC files, wait for conversion before running AnalyzeJob [#184792253]

### PR DESCRIPTION
**Pull request by alaroia-cfa**
I believe this will fix many/all of the crashes involving AnalyzeJob.

### CODE REVIEW

File: app/models/document.rb [hunk 1]
```diff
@@ -64,6 +64,13 @@ class Document < ApplicationRecord
   scope :active, ->() { where(archived: false) }
   scope :archived, ->() { where(archived: true) }
 
+  after_save do
+    if upload.present? && upload.blob.present? && upload.blob.filename.extension_without_delimiter.downcase == "heic" && !upload.blob.persisted?
```
Comment (tgrathwell) [inline app/models/document.rb:68]: maybe `upload.filename.extension_without_delimiter.downcase` can be `Document#is_heic?`

File: app/models/document.rb [hunk 2]
```diff
@@ -64,6 +64,13 @@ class Document < ApplicationRecord
...
+      # Skip AnalyzeJob when initially creating .heic files, since we will analyze them after JPG conversion
+      upload.blob.analyzed = true
```
Comment (tgrathwell) [inline app/models/document.rb:70]: feels sneaky but if it works it works

File: spec/models/document_spec.rb [hunk 3]
```diff
@@ -217,25 +217,34 @@
...
-        allow(HeicToJpgJob).to receive(:perform_later)
+      before do
+        allow(HeicToJpgJob).to receive(:perform_later).and_call_original
+      end
 
+      it "creates a job to convert the file to jpg and delays ActiveStorage::AnalyzeJob until then" do
+        document = build :document, upload_path: Rails.root.join("spec", "fixtures", "files", "IMG_4851.HEIC")
         document.save!
 
+        expect(ActiveJob::Base.queue_adapter.enqueued_jobs.map { |x| x["job_class"] }).not_to include("ActiveStorage::AnalyzeJob")
```
Comment (tgrathwell) [inline spec/models/document_spec.rb:231]: could we instead assert here that the `enqueued_jobs` is just a `[HeicToJpgJob]`

**Review by tgrathwell — APPROVED** [review]

File: app/models/document.rb (hunk 2 above)
Comment (alaroia-cfa) [reply app/models/document.rb]: 👟

File: spec/models/document_spec.rb (hunk 3 above)
Comment (alaroia-cfa) [reply spec/models/document_spec.rb]: done, yay

File: app/models/document.rb (hunk 1 above)
Comment (alaroia-cfa) [reply app/models/document.rb]: yes but with so many ampersands you'll be amazed

**Pull request merged**